src/main/java/myjournal/Parser.java

```diff
@@ -0,0 +1,292 @@
+                return ui.findTaskPrint(Parser.parseFind(line, tasks));
+            case "list":
+                return Parser.parseList(tasks);
+            case "todo":
+                tasks.addTask(Parser.parseTodo(line));
+                return ui.taskAddPrint(tasks);
+            case "event":
+                tasks.addTask(Parser.parseEvent(line));
+                return ui.taskAddPrint(tasks);
+            case "deadline":
+                tasks.addTask(Parser.parseDeadline(line));
+                return ui.taskAddPrint(tasks);
+            default:
+                throw new InvalidTypeException("OOPS!!! Please put either todo/event/deadline!");
+            }
+        } catch (InvalidTypeException e) {
+            return e.toString();
+        } catch (InvalidTaskNumberException e) {
+            return e.toString();
+        } catch (EmptyDescriptionException e) {
+            return e.toString();
+        } catch (InvalidWordException e) {
+            return e.toString();
+        } catch (DateTimeParseException exception) {
+            return exception.toString();
+        }
+    }
+
+    /**
+     * Parses user's input for the command "find".
+     *
+     * @param line The current line that is being parsed.
+     * @param tasks The list of tasks.
+     * @return A task which has been marked as done.
+     */
+    public static TaskList parseFind(Scanner line, TaskList tasks) {
+        if (!line.hasNext()) {
+            throw new EmptyDescriptionException("OOPS!!! Please tell me which word you want to find!!");
+        }
+        String find = line.next();
+        if (line.hasNext()) {
+            throw new InvalidWordException("Please enter only one word!!");
+        }
+        TaskList newList = new TaskList();
+        for (int i = 0; i < tasks.getSize(); i++) {
+            Task currTask = tasks.getTask(i);
+            Scanner taskName = new Scanner(currTask.getTaskName());
+            while (taskName.hasNext()) {
+                if (taskName.next().equals(find)) {
+                    newList.addTask(currTask);
+                }
+            }
+        }
+        return newList;
+    }
+
+    /**
+     * Parses user's input for the command "done".
+     *
+     * @param line The current line that is being parsed.
+     * @param tasks The list of tasks.
+     * @return A task which has been marked as done.
+     */
+    public static Task parseDone(Scanner line, TaskList tasks) {
+        if (!line.hasNextInt()) {
+            throw new InvalidTaskNumberException("OOPS!!! Please specify the task "
+                    + "that needs to be marked as done!");
+        }
+        int index = line.nextInt() - 1;
+        if (index >= tasks.getSize() || index < 0 || tasks.getTask(index) == null) {
+            throw new InvalidTaskNumberException("OOPS!!! Please enter a valid task number!");
+        }
+        tasks.getTask(index).setState(true);
+        return tasks.getTask(index);
+    }
```

Comment:
Since the Exception thrown is not caught in the method body, shouldn't the method signature include `throws InvalidTaskNumberException` or `throws EmptyDescriptionException, InvalidWordException`

Reply:
Perhaps this could apply to parseDelete too?